server/client_fake.go

```diff
@@ -38,19 +37,23 @@ func (client *ClientFake) NodeSendStats(nodeKey string, sessionStats []dto.Sessi
 	return nil
 }
 
-func (client *ClientFake) SessionCreate(nodeKey string) (session dto.Session, err error) {
+func (client *ClientFake) Proposals(nodeKey string) (proposals []dto_discovery.ServiceProposal, err error) {
+	log.Info(MYSTERIUM_API_LOG_PREFIX, "Fake proposals requested for node_key: ", nodeKey)
 	if proposal, ok := client.proposalsByProvider[nodeKey]; ok {
-		session = dto.Session{
-			Id:              nodeKey + "-session",
-			ServiceProposal: proposal,
-		}
-		return
+		proposals = []dto_discovery.ServiceProposal{proposal}
+	} else {
+		proposals = []dto_discovery.ServiceProposal{}
```

Comment:
shouldn't we return an error here instead of new empty proposal?

Reply:
If client by contract returns zero or more proposals then it's ok to return zero propsals, caller will decide if it is error (connection manager) or its normal( /proposals tequilapi endpoint)

Reply:
he decided to push this error to upper logic level

Reply:
Yeah - `/v1/proposals` endpoint works this way:
- By default, it returns all proposals.
- It allows filtering by some values. If filtering doesn't return any matches, it doesn't mean that it's an error - it means that there are no objects by given filters.

Reply:
+1